lib/tmdb/client.rb

```diff
@@ -20,9 +20,47 @@ def get_movie_search_results(movie_title)
         )
       end
 
+      def get_advanced_movie_search_results(params)
+        searched_terms = SearchParamParser.parse_movie_params_for_display(params)
+        data = if params[:actor_name].present?
+         person_id = request(:person_search, query: params[:actor_name])&.dig(:results)&.first&.dig(:id)
+         return OpenStruct.new(not_found_message: "No results for actor '#{params[:actor_name]}'.") if person_id.blank?
+
+          request(:discover_search, params.except(:actor_name).merge(people: person_id))
+        else
+          request(:discover_search, params.except(:actor_name))
+        end
+
+        movie_results = data.dig(:results)
+        return OpenStruct.new(not_found_message: "No results for #{searched_terms}.") if movie_results.blank?
+
+        movies = MovieSearch.parse_results(movie_results)
+        total_pages = data.fetch(:total_pages)
+        current_page = data[:page]
+
+        OpenStruct.new(
+          searched_terms: searched_terms,
+          searched_params: {
+            actor_name: params[:actor_name],
+            company: params[:company],
+            date: params[:date],
+            genre: params[:genre],
+            mpaa_rating: params[:mpaa_rating],
+            sort_by: params[:sort_by],
+            timeframe: params[:timeframe],
+            year: params[:year]
+          },
```

Comment:
I'm returning 2 sets of params for use in the view.
`searched_terms`: this is the string that looks like "search results for Will Smith Action movies rated R after 1990" 
`searched_params`: just passes back what was originally searched. This is used in the pagination links.